nomos-services/network/src/backends/libp2p/swarm.rs

```diff
@@ -92,6 +131,85 @@ impl SwarmHandler {
                 tracing::debug!("Got message with id: {id} from peer: {peer_id}");
                 log_error!(self.events_tx.send(Event::Message(message)));
             }
+            // identify events (used for peer discovery)
+            SwarmEvent::Behaviour(BehaviourEvent::Identify(identify::Event::Received {
+                peer_id,
+                info,
+                ..
+            })) => {
+                tracing::debug!(
+                    "Identified peer {} with addresses {:?}",
+                    peer_id,
+                    info.listen_addrs
+                );
+
+                let behaviour = self.swarm.swarm_mut().behaviour_mut();
+
+                if info.protocols.iter().any(|p| *p == KAD_PROTOCOL_NAME) {
+                    // we need to add the peer to the kademlia routing table
+                    // in order to enable peer discovery
+                    for addr in &info.listen_addrs {
+                        behaviour.kademlia_add_address(peer_id, addr.clone());
+                    }
+                }
+            }
+
+            SwarmEvent::Behaviour(BehaviourEvent::Kademlia(
+                kad::Event::OutboundQueryProgressed {
```

Comment:
Now with more events, it could be handled per behaviour in one place, similar like it's done here: https://github.com/logos-co/nomos/blob/master/nomos-da/network/core/src/swarm/validator.rs#L265-L267